src/scripts/import-needs-assessment-data/add-categories.ts

```diff
@@ -0,0 +1,248 @@
+import { STRAPI_ENV } from "../strapi-env";
+import {
+  Category,
+  NeedAssessment,
+  CategoryUploadWorkflow,
+  UploadWorkflowStatus,
+  CategoryUploadWorkflowResults,
+} from "./types.d";
+
+/*  Add Categories from Needs Assessment Data
+ * ------------------------------------------------------- */
+export async function addCategories(
+  data: NeedAssessment[],
+): Promise<Category[]> {
+  console.log("Adding Product.Categories from the Needs Assessment data ...");
+
+  const uniqueCategories = consolidateCategories(data);
+
+  const results = await Promise.allSettled<CategoryUploadWorkflow>(
+    uniqueCategories.map((category) => {
+      return new Promise<CategoryUploadWorkflow>((resolve, _reject) => {
+        resolve({
+          data: {
+            category,
+          },
+          orig: category,
+          status: UploadWorkflowStatus.PROCESSING,
+          logs: [],
+        });
+      })
+        .then(parseCategory)
+        .then(getCategory)
+        .then(uploadCategory);
+    }),
+  );
+
+  // { "SUCCESS": [], "ALREADY_EXITS": [], ...}
+  const resultsMap: CategoryUploadWorkflowResults = Object.keys(
```

Comment:
Can we simplify this slightly?

We might be able to use some ternary expression to reduce the if statements and map the data directly.

Something like:
``` 
const status = workflowResult.status || UploadWorkflowStatus.OTHER;
const workflowResult = isFulfilled(result) ? result.value : result.reason; 
```

And then from there we can create an object/dictionary to story the results
```
resultsObj[status] = status;
resultsObj[status].push(workflowResult);
```
and then we can add that object as needed

This is just some pseudocode form my head but I think we might be able to simply this whole block maybe.

Reply:
@phageogram addressed this in [PR#227](https://github.com/distributeaid/aggregated-public-information/pull/227). Would love your review on that to confirm it covers what you had in mind!